aws-lc-rs/src/unstable/kem.rs

```diff
@@ -0,0 +1,339 @@
+};
+
+/// NIST Round 3 submission of the Kyber-1024 algorithm.
+const KYBER1024_R3: Algorithm<AlgorithmId> = Algorithm {
+    id: AlgorithmId::Kyber1024_R3,
+    secret_key_size: KYBER1024_R3_SECRET_KEY_LENGTH,
+    public_key_size: KYBER1024_R3_PUBLIC_KEY_LENGTH,
+    ciphertext_size: KYBER1024_R3_CIPHERTEXT_LENGTH,
+    shared_secret_size: KYBER1024_R3_SHARED_SECRET_LENGTH,
+};
+
+/// Identifier for an unstable KEM algorithm.
+#[allow(non_camel_case_types)]
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AlgorithmId {
+    /// NIST Round 3 submission of the Kyber-512 algorithm.
+    Kyber512_R3,
+
+    /// NIST Round 3 submission of the Kyber-768 algorithm.
+    Kyber768_R3,
+
+    /// NIST Round 3 submission of the Kyber-1024 algorithm.
+    Kyber1024_R3,
+}
+
+impl crate::kem::AlgorithmIdentifier for AlgorithmId {
+    #[inline]
+    fn nid(self) -> i32 {
+        match self {
+            AlgorithmId::Kyber512_R3 => NID_KYBER512_R3,
+            AlgorithmId::Kyber768_R3 => NID_KYBER768_R3,
+            AlgorithmId::Kyber1024_R3 => NID_KYBER1024_R3,
+        }
+    }
+}
+
+impl crate::sealed::Sealed for AlgorithmId {}
+
+/// Retrieve an unstable KEM [`Algorithm`] using the [`AlgorithmId`] specified by `id`.
+/// May return [`Option::None`] if support for the algorithm has been removed from the unstable module.
+#[must_use]
+pub const fn get_algorithm(id: AlgorithmId) -> Option<&'static Algorithm<AlgorithmId>> {
+    match id {
+        AlgorithmId::Kyber512_R3 => Some(&KYBER512_R3),
+        AlgorithmId::Kyber768_R3 => Some(&KYBER768_R3),
+        AlgorithmId::Kyber1024_R3 => Some(&KYBER1024_R3),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        error::KeyRejected,
+        kem::{DecapsulationKey, EncapsulationKey},
+    };
+
+    use super::{get_algorithm, AlgorithmId, KYBER1024_R3, KYBER512_R3, KYBER768_R3};
+
+    #[cfg(private_api)]
+    macro_rules! kem_kat_test {
+        ($file:literal, $alg:expr) => {
+            use crate::{
+                aws_lc::{
+                    pq_custom_randombytes_init_for_testing,
+                    pq_custom_randombytes_use_deterministic_for_testing,
+                },
+                test, test_file,
+            };
+            test::run(test_file!($file), |_section, test_case| {
+                let seed = test_case.consume_bytes("seed");
+                let public_key_bytes = test_case.consume_bytes("pk");
+                let secret_key_bytes = test_case.consume_bytes("sk");
+                let ciphertext_bytes = test_case.consume_bytes("ct");
+                let shared_secret_bytes = test_case.consume_bytes("ss");
+
+                // Set randomness generation in deterministic mode.
+                unsafe {
+                    pq_custom_randombytes_use_deterministic_for_testing();
+                    pq_custom_randombytes_init_for_testing(seed.as_ptr());
+                }
+
+                let priv_key = PrivateKey::generate($alg).unwrap();
```

Comment:
At some point these tests became broken:
```
   Compiling aws-lc-sys v0.12.1 (/Users/justsmth/repos/aws-lc-rs/aws-lc-sys)
error[E0433]: failed to resolve: use of undeclared type `PrivateKey`
   --> aws-lc-rs/src/unstable/kem.rs:171:32
    |
171 |                 let priv_key = PrivateKey::generate($alg).unwrap();
    |                                ^^^^^^^^^^ use of undeclared type `PrivateKey`
...
193 |         kem_kat_test!("../../tests/data/kyber512r3.txt", &KYBER512_R3);
    |         -------------------------------------------------------------- in this macro invocation
````

Reply:
See previous comment about this being removed.